docs/src/systems/implicit_incompressible_sph.md

```diff
@@ -0,0 +1,291 @@
+```
+
+Because interactions are local, limited to particles within the kernel support defined by
+the smoothing length, each particle's pressure $p_i$ depends only on its own value and
+nearby particles.
+Consequently, the matrix A is sparse (most entries are zero).
+
+The diagonal elements ``a_ii`` get computed and stored at the beginning of the simulation step
+and remain unchanged throughout the relaxed Jacobi iterations. The same applies for the
+$d_{ii}$. The coefficients $d_{ij}$ are computed, but not stored, as part of the calculation
+of $a_{ii}$.
+During the Jacobi iterations, two loops over all particles are performed:
+The first updates the values of $\sum_j d_{ij}$ and the second computes the updated
+pressure values $p_i^{l+1}$.
+
+The final pressure update follows the equation of the relaxed Jacobi scheme as stated above, with two
+exceptions:
+#### 1. Pressure clamping
+To avoid negative pressure values, one can enable pressure clamping. In this case, the
+pressure update is given by
+
+```math
+p_i^{l+1} = \max \left(0, (1-\omega) p_i^l + \omega \frac{\rho_0 - \rho_i^{\text{adv}} - \sum_{j \neq i} a_{ij}p_j^l}{a_{ii}}\right).
+```
+
+#### 2. Small diagonal elements
+If the diagonal element $a_{ii}$ becomes too small or even zero, the simulation may become
+unstable. This can occur, for example, if a particle is isolated and receives little or no
+influence from neighboring particles, or due to numerical cancellations. In such cases, the
+updated pressure value is set to zero.
+
+There are also other options, like setting $a_{ii}$ to the threshold value if it is beneath
+and then update with the usual equation, or just don't update the pressure value at all, and
+continue with the old value. The numerical error introduced by this technique remains small,
+as only isolated or almost isolated particles are affected.
+
+
+## Boundary Handling
+
+The previously introduced formulation only considered interactions between fluid particles
+and neglected interactions between fluid and boundary particles. To account for boundary
+interactions, a few modifications to the previous equations are required.
+
+First, the discretized form of the continuity equation must be adapted for the case in which
+a neighboring particle is a boundary particle. From now on, we distinguish between
+neighboring fluid particles (indexed by $j$) and neighboring boundary particles (indexed by
+$b$).
+
+The updated discretized continuity equation becomes:
+
+```math
+\frac{\rho_i(t + \Delta t) - \rho_i(t)}{\Delta t} = \sum_j m_j \bm{v}_{ij}(t+\Delta t) \nabla W_{ij} + \sum_b m_b \bm{v}_{ib}(t+\Delta t) \nabla W_{ib}.
+```
+
+Since boundary particles do not have an own velocity the difference between the fluid
+particles velocity and the boundary particles velocity simplifies to just the fluid
+particles velocity $\bm{v}_{ib}(t+\Delta t) = \bm{v}_{i}(t+\Delta t)$
```

Comment:
```suggestion
Since boundary particles have zero velocity, the difference between the fluid
particle's velocity and the boundary particle's velocity simplifies to just the fluid
particle's velocity $\bm{v}_{ib}(t+\Delta t) = \bm{v}_{i}(t+\Delta t)$.
```